src/test/java/org/prebid/server/analytics/reporter/greenbids/GreenbidsAnalyticsReporterTest.java

```diff
@@ -0,0 +1,468 @@
+        verify(mockResponse).getStatusCode();
+    }
+
+    @Test
+    public void shouldReceiveValidResponseOnAuctionContextForVideo() {
+        // given
+        final Video video = setUpVideo();
+        final Imp imp = Imp.builder()
+                .video(video)
+                .build();
+        final AuctionContext auctionContext = setupAuctionContext(imp);
+        event = AuctionEvent.builder()
+                .auctionContext(auctionContext)
+                .bidResponse(auctionContext.getBidResponse())
+                .build();
+
+        final HttpClientResponse mockResponse = mock(HttpClientResponse.class);
+        when(mockResponse.getStatusCode()).thenReturn(202);
+        when(httpClient.post(anyString(), any(MultiMap.class), anyString(), anyLong()))
+                .thenReturn(Future.succeededFuture(mockResponse));
+
+        // when
+        final Future<Void> result = target.processEvent(event);
+
+        // then
+        assertThat(result.succeeded()).isTrue();
+        verify(httpClient).post(anyString(), any(MultiMap.class), anyString(), anyLong());
+        verify(mockResponse).getStatusCode();
+    }
+
+    @Test
+    public void shouldFailWhenBidResponseIsNull() {
+        // given
+        final AuctionContext auctionContext = setUpAuctionContextWithNoBidResponse();
+        event = AuctionEvent.builder()
+                .auctionContext(auctionContext)
+                .bidResponse(auctionContext.getBidResponse())
+                .build();
+
+        // when
+        final Future<Void> result = target.processEvent(event);
+
+        // then
+        assertThat(result.failed()).isTrue();
+        assertThat(result.cause())
+                .hasMessageStartingWith("Bid response or auction context cannot be null");
+    }
+
+    @Test
+    public void shouldFailWhenAuctionContextIsNull() {
+        // given
+        final AuctionEvent event = mock(AuctionEvent.class);
+        when(event.getAuctionContext()).thenReturn(null);
+
+        // when
+        final Future<Void> result = target.processEvent(event);
+
+        // then
+        assertThat(result.failed()).isTrue();
+        assertThat(result.cause())
+                .hasMessageStartingWith("Bid response or auction context cannot be null");
+    }
+
+    @Test
+    public void shouldFailOnEncodeException() {
+        // given
+        final Banner banner = setUpBanner();
+        final Imp imp = Imp.builder()
+                .banner(banner)
+                .build();
+        final AuctionContext auctionContext = setupAuctionContext(imp);
+        event = AuctionEvent.builder()
+                .auctionContext(auctionContext)
+                .bidResponse(auctionContext.getBidResponse())
+                .build();
+
+        final JacksonMapper mockJacksonMapper = mock(JacksonMapper.class);
+        doThrow(new EncodeException("Failed to encode as JSON")).when(mockJacksonMapper).encodeToString(any());
```

Comment:
that's an overhead, just create an invalid JsonNode for the test purpose

then you don't need to mock the mapper and re-create the target

Reply:
discussed on Slack https://prebid.slack.com/archives/D0755STJUU9/p1717503988751439

Reply:
I agree that it's better to craft an invalid input for test purpose instead of relying on an artificial mock.
However, as @EvgeniiMunin said on Slack, it's not possible to raise the `EncodeException` with invalid input (the invalid input will fallback to `null`, making the `CommonMessage`, maybe incomplete, but fully encodable).
This leads me to think that you try to test an impossible case, and thus such test is not required.
I don't know the best practice. If such kind of error is not expected, maybe you could remove the `catch` block handling it.